internal/settings/linode.go

```diff
@@ -0,0 +1,321 @@
+package settings
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/qdm12/ddns-updater/internal/constants"
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/regex"
+	netlib "github.com/qdm12/golibs/network"
+)
+
+type linode struct {
+	domain    string
+	host      string
+	ipVersion models.IPVersion
+	token     string
+}
+
+func NewLinode(data json.RawMessage, domain, host string, ipVersion models.IPVersion,
+	_ bool, _ regex.Matcher) (s Settings, err error) {
+	extraSettings := struct {
+		Token string `json:"token"`
+	}{}
+	if err := json.Unmarshal(data, &extraSettings); err != nil {
+		return nil, err
+	}
+	l := &linode{
+		domain:    domain,
+		host:      host,
+		ipVersion: ipVersion,
+		token:     extraSettings.Token,
+	}
+	if err := l.isValid(); err != nil {
+		return nil, err
+	}
+	return l, nil
+}
+
+func (l *linode) isValid() error {
+	if len(l.token) == 0 {
+		return ErrEmptyToken
+	}
+	return nil
+}
+
+func (l *linode) String() string {
+	return toString(l.domain, l.host, constants.LUADNS, l.ipVersion)
+}
+
+func (l *linode) Domain() string {
+	return l.domain
+}
+
+func (l *linode) Host() string {
+	return l.host
+}
+
+func (l *linode) DNSLookup() bool {
+	return true
+}
+
+func (l *linode) IPVersion() models.IPVersion {
+	return l.ipVersion
+}
+
+func (l *linode) BuildDomainName() string {
+	return buildDomainName(l.host, l.domain)
+}
+
+func (l *linode) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    models.HTML(fmt.Sprintf("<a href=\"http://%s\">%s</a>", l.BuildDomainName(), l.BuildDomainName())),
```

Comment:
The config json asks to add the domain name (example.com), but this line concatenates the host and domain name (subdomain.example.com) under the header "Domain" - is this intentional, given that we have an extra column for host directly after this one, or should the domain in this column match the domain configured in the config file?

Reply:
Yes this is intentional, because a particular row is about host + domain, and can only be about the domain only if the host is set to `@`. But you're right it's a tiny bit confusing, I'll take it into account when I do a v2 UI 👍

Reply:
Thanks!